plonky2/src/hash/keccak.rs

```diff
@@ -56,7 +56,7 @@ impl<F: RichField, const N: usize> Hasher<F> for KeccakHash<N> {
     type Hash = BytesHash<N>;
     type Permutation = KeccakPermutation;
 
-    fn hash(input: &[F], _pad: bool) -> Self::Hash {
```

Comment:
This was ignoring the flag, which I guess was fine since the `keccak` call should pad. We might be able to skip padding here, but I don't think we should since we need consistency with the EVM's `KECCAK256` instruction, which I believe will always pad.

After this change, `hash_no_pad` will pad, and `hash_pad` will double-pad. I guess I should override `hash_pad` to call `hash_no_pad`, so that both will pad once, as before. Does that sound good?

Reply:
Sounds good to me.

Reply:
Sounds good to me.